### PR TITLE
feat: Regenerate with latest generator and api defs

### PR DESCRIPTION
**Pull request by Mikemosca**
BREAKING CHANGE: Version 2 Major release

### CODE REVIEW

File: lib/ibm_watson/visual_recognition_v4.rb [hunk 1]
```diff
@@ -280,7 +288,8 @@ def update_collection(collection_id:, name: nil, description: nil)
 
       data = {
         "name" => name,
-        "description" => description
+        "description" => description,
+        "training_status" => training_status
```
Comment (mediumTaj) [inline lib/ibm_watson/visual_recognition_v4.rb:292]: These should be readonly and not generated. need to look at generator

File: lib/ibm_watson/visual_recognition_v4.rb [hunk 2]
```diff
@@ -181,7 +181,8 @@ def create_collection(name: nil, description: nil)
...
```
Comment (mediumTaj) [inline lib/ibm_watson/visual_recognition_v4.rb:185]: These should be readonly and not generated. need to look at generator

File: lib/ibm_watson/visual_recognition_v4.rb (hunk 1 above)
Comment (Mikemosca) [reply lib/ibm_watson/visual_recognition_v4.rb:292]: Just to clarify, Are you saying I need to add hand edits to remove training_status?
It is also passed in so remove it as a parameter to?

    # @param training_status [TrainingStatus] Training status information for the collection.
    # @return [IBMCloudSdkCore::DetailedResponse] A `IBMCloudSdkCore::DetailedResponse` object representing the response.
    def update_collection(collection_id:, name: nil, description: nil, training_status: nil)

Comment (mediumTaj) [reply lib/ibm_watson/visual_recognition_v4.rb:292]: Yes, training status should be read only and not a param in the method signature. We will discuss this in standup.

File: lib/ibm_watson/visual_recognition_v3.rb [hunk 3]
```diff
@@ -32,33 +39,27 @@ class VisualRecognitionV3 < IBMCloudSdkCore::BaseService
...
-    #   ready for a later version.
+    # @option args version [String] Release date of the API version you want to use. Specify dates in YYYY-MM-DD
+    #   format. The current version is `2018-03-19`.
     # @option args service_url [String] The base service URL to use when contacting the service.
     #   The base service_url may differ between IBM Cloud regions.
     # @option args authenticator [Object] The Authenticator instance to be configured for this service.
     # @option args service_name [String] The name of the service to configure. Will be used as the key to load
     #   any external configuration, if applicable.
     def initialize(args = {})
+      warn "On 1 December 2021, Visual Recognition will no longer be available. For more information, see the README."
```
Comment (mediumTaj) [inline lib/ibm_watson/visual_recognition_v3.rb:56]: Can we add the url here? https://github.com/watson-developer-cloud/ruby-sdk/tree/master#visual-recognition-deprecation

File: lib/ibm_watson/visual_recognition_v4.rb [hunk 4]
```diff
@@ -31,33 +38,27 @@ class VisualRecognitionV4 < IBMCloudSdkCore::BaseService
...
-    #   ready for a later version.
+    # @option args version [String] Release date of the API version you want to use. Specify dates in YYYY-MM-DD
+    #   format. The current version is `2019-02-11`.
     # @option args service_url [String] The base service URL to use when contacting the service.
     #   The base service_url may differ between IBM Cloud regions.
     # @option args authenticator [Object] The Authenticator instance to be configured for this service.
     # @option args service_name [String] The name of the service to configure. Will be used as the key to load
     #   any external configuration, if applicable.
     def initialize(args = {})
+      warn "On 1 December 2021, Visual Recognition will no longer be available. For more information, see the README."
```
Comment (mediumTaj) [inline lib/ibm_watson/visual_recognition_v4.rb:55]: can we add the url here? https://github.com/watson-developer-cloud/ruby-sdk/tree/master#visual-recognition-deprecation

**Review by mediumTaj — APPROVED** [review]
👍 looks good! just a couple of comments

File: lib/ibm_watson/personality_insights_v3.rb [hunk 5]
```diff
@@ -49,33 +62,27 @@ class PersonalityInsightsV3 < IBMCloudSdkCore::BaseService
...
-    #   ready for a later version.
+    # @option args version [String] Release date of the version of the API you want to use. Specify dates in
+    #   YYYY-MM-DD format. The current version is `2017-10-13`.
     # @option args service_url [String] The base service URL to use when contacting the service.
     #   The base service_url may differ between IBM Cloud regions.
     # @option args authenticator [Object] The Authenticator instance to be configured for this service.
     # @option args service_name [String] The name of the service to configure. Will be used as the key to load
     #   any external configuration, if applicable.
     def initialize(args = {})
+      warn "On 1 December 2021, Personality Insights will no longer be available. For more information, see the README."
```
[the remark this answers is not in the document]
Comment (Mikemosca) [reply lib/ibm_watson/personality_insights_v3.rb]: yup

File: lib/ibm_watson/visual_recognition_v3.rb (hunk 3 above)
Comment (Mikemosca) [reply lib/ibm_watson/visual_recognition_v3.rb]: yup

File: lib/ibm_watson/visual_recognition_v4.rb (hunk 4 above)
Comment (Mikemosca) [reply lib/ibm_watson/visual_recognition_v4.rb]: yup

**Review by mediumTaj — APPROVED** [review]
👍 looks good!

**Review by mamoonraja — DISMISSED** [review]
LGTM!

**Review by mediumTaj — APPROVED** [review]

**Review by mediumTaj — APPROVED** [review]
👍 looks good!

**Pull request merged**